specs/chain/hs/src/Cardano/Spec/Chain/STS/Rule/Chain.hs

```diff
@@ -164,17 +143,28 @@ instance HasTrace CHAIN where
     -- those values here. The upper bound is arbitrary though.
     mHSz <- Gen.integral (Range.constant 0 4000000)
     mBSz <- Gen.integral (Range.constant 0 4000000)
+
     mTxSz <- Gen.integral (Range.constant 0 4000000)
     mPSz <- Gen.integral (Range.constant 0 4000000)
-    -- The delegation liveness parameter is arbitrarily determined.
-    d <- SlotCount <$> Gen.integral (Range.linear 0 10)
-    -- The size of the rolling widow is arbitrarily determined.
-    w <- Gen.integral (Range.linear 0 10)
+
+    -- The size of the rolling window should be equal to k.
+    -- TODO: we need to adapt the formal specs to reflect this.
+    w <- Gen.integral (Range.linear 1 10)
+
+    -- TODO: The delegation liveness parameter is equal to `2 * k`.
+    -- We should adapt the specs to reflect this.
+    d <- pure $! SlotCount $ 2 * fromIntegral w
+
     -- The percentage of the slots will typically be between 1/5 and 1/4,
     -- however we want to stretch that range a bit for testing purposes.
-    t <- Gen.double (Range.constant (1/6) (1/3))
-    -- The slots per-epoch is arbitrarily determined.
-    spe <- SlotCount <$> Gen.integral (Range.linear 1 1000)
+    t <- pure (1/5) -- Gen.double (Range.constant (1/6) (1/3))
+    -- TODO: we make this a constant till we solve the problem with the number
+    -- of byzantineNodes being a constant in the implementation.
+
+    -- TODO: at the moment the number of slots per epoch is computed from 'k':
+    -- spe = k * 10, and k = w. So we should adapt the formal specs to account
+    -- for this.
```

Comment:
Should we? Is the intention for `k` also to get larger when we increase the slots per epoch?

Reply:
The thing is that we're specifying Byron, where `k` and `slots-per-epoch` are related in this way. In the current spec I did not relate these two things, but then the validation won't pass. So I don't know whether it makes sense to make the spec for the old system more general, but then have to perform these hacks in the generators.

For Shelley we shouldn't have `slots-per-epoch` depending on `k` of course.